src/plugins/hetero/src/plugin.cpp

```diff
@@ -0,0 +1,289 @@
+}
+
+ov::hetero::Plugin::DeviceMetaInformationMap ov::hetero::Plugin::GetDevicePlugins(const std::string& targetFallback,
+                                                                                  const ov::AnyMap& properties) const {
+    auto fallbackDevices = ov::DeviceIDParser::get_hetero_devices(targetFallback);
+    DeviceMetaInformationMap metaDevices;
+    for (auto&& deviceName : fallbackDevices) {
+        auto itPlugin = metaDevices.find(deviceName);
+        if (metaDevices.end() == itPlugin) {
+            metaDevices[deviceName] = get_core()->get_supported_property(deviceName, properties);
+        }
+    }
+    return metaDevices;
+}
+
+ov::SupportedOpsMap ov::hetero::Plugin::query_model(const std::shared_ptr<const ov::Model>& model,
+                                                           const ov::AnyMap& properties) const {
+    OV_ITT_SCOPED_TASK(itt::domains::Hetero, "Plugin::query_model");
+
+    Configuration fullConfig{properties, m_cfg, false};
+    
+    OPENVINO_ASSERT(model, "OpenVINO Model is empty!");
+
+    std::string fallbackDevicesStr = fullConfig.device_priorities;
+    
+    DeviceMetaInformationMap metaDevices = GetDevicePlugins(fallbackDevicesStr, fullConfig.GetDeviceConfig());
+
+    std::map<std::string, ov::SupportedOpsMap> queryResults;
+    for (auto&& metaDevice : metaDevices) {
+        const auto& deviceName = metaDevice.first;
+        const auto& device_config = metaDevice.second;
+        queryResults[deviceName] = get_core()->query_model(model, deviceName, device_config);
+    }
+
+    //  WARNING: Here is devices with user set priority
+    auto fallbackDevices = ov::DeviceIDParser::get_hetero_devices(fallbackDevicesStr);
+
+    ov::SupportedOpsMap res;
+    for (auto&& deviceName : fallbackDevices) {
+        for (auto&& layerQueryResult : queryResults[deviceName]) {
+            res.emplace(layerQueryResult);
+        }
+    }
+
+    return res;
+}
+
+void ov::hetero::Plugin::set_property(const ov::AnyMap& properties) {
+    m_cfg = Configuration{properties, m_cfg};
+}
+
+ov::Any ov::hetero::Plugin::get_property(const std::string& name, const ov::AnyMap& arguments) const {
+    const auto& add_ro_properties = [](const std::string& name, std::vector<ov::PropertyName>& properties) {
+        properties.emplace_back(ov::PropertyName{name, ov::PropertyMutability::RO});
+    };
+
+    const auto& default_ro_properties = []() {
+        std::vector<ov::PropertyName> ro_properties{ov::supported_properties,
+                                                    ov::device::full_name,
+                                                    ov::device::capabilities,
+                                                    ov::caching_properties};
+                                                    //ov::available_devices,
+                                                    //ov::device::architecture,
+                                                    //ov::range_for_async_infer_requests
```

Comment:
Did we have these properties before?

Reply:
No, it was copy/paste from template plugin to align hetero with new format